apps/entropy-tester/src/index.ts

```diff
@@ -0,0 +1,168 @@
+      (contract) => contract.chain.getId() == config["chain-id"],
+    );
+    if (contracts.length === 0) {
+      throw new Error(
+        `Can not find the contract for chain ${config["chain-id"]}, check contract manager store.`,
+      );
+    }
+    if (contracts.length > 1) {
+      throw new Error(
+        `Multiple contracts found for chain ${config["chain-id"]}, check contract manager store.`,
+      );
+    }
+    loadedConfigs.push({ contract: contracts[0], interval });
+  }
+  return loadedConfigs;
+}
+
+async function testLatency(
+  contract: EvmEntropyContract,
+  privateKey: PrivateKey,
+  logger: Logger,
+) {
+  const provider = await contract.getDefaultProvider();
+  const userRandomNumber = contract.generateUserRandomNumber();
+  const requestResponse = await contract.requestRandomness(
+    userRandomNumber,
+    provider,
+    privateKey,
+    true, // with callback
+  );
+  // Read the sequence number for the request from the transaction events.
+  const sequenceNumber = parseInt(
+    requestResponse.events.RequestedWithCallback.returnValues.sequenceNumber,
+  );
+  logger.info(
+    { sequenceNumber, txHash: requestResponse.transactionHash },
+    `Request submitted`,
+  );
+
+  const startTime = Date.now();
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+    const request = await contract.getRequest(provider, sequenceNumber);
+    logger.debug(request);
+
+    if (parseInt(request.sequenceNumber) === 0) {
+      // 0 means the request is cleared
+      const endTime = Date.now();
+      logger.info(
+        { sequenceNumber, latency: endTime - startTime },
+        `Successful callback`,
+      );
+      break;
+    }
+    if (Date.now() - startTime > 60000) {
+      logger.error(
+        { sequenceNumber },
+        "Timeout: 60s passed without the callback being called",
+      );
+      break;
+    }
+  }
+}
+
+yargs(hideBin(process.argv))
+  .parserConfiguration({
+    "parse-numbers": false,
+  })
+  .command({
+    command: "run",
+    describe: "run the tester until manually stopped",
+    builder: {
+      validate: {
+        description: "Only validate the configs and exit",
+        type: "boolean",
+        default: false,
+        required: false,
+      },
+      config: {
+        description: "Yaml config file",
+        type: "string",
+        required: true,
+      },
+      "private-key": {
+        type: "string",
+        required: true,
+        description:
+          "Path to the private key to sign the transactions with. Should be hex encoded",
+      },
+    },
+    handler: async (argv: any) => {
+      const logger = pino();
+      const configs = loadConfig(argv.config);
+      if (argv.validate) {
+        logger.info("Config validated");
+        return;
+      }
+      const privateKey = toPrivateKey(
+        fs
+          .readFileSync(argv["private-key"], "utf-8")
```

Comment:
I suggest using `fs/promises` and making this async/await.  `readFileSync` is largely unnecessary now that async/await works, and using it at all is kind of a bad habit because it really only serves to mask the async nature of what you're doing, and in some cases it can unintentionally cause major issues because code gets deferred that doesn't need to be.

The only case I recommend using `readFileSync` or other such sync functions is if you absolutely must add code like this in a place where the code is not designed for async, and converting it would be a huge pain.  But here, that's not the case, you're already in an async function.